doc/api/quic.md

```diff
@@ -2411,15 +2418,47 @@ added: REPLACEME
 
 Set to `true` if the `QuicStream` is unidirectional.
 
+## Additional Notes
+
+### Custom DNS Lookup Functions
+
+By default, the QUIC implementation uses the `dns` module's
+[promisified version of `lookup()`][] to resolve domains names
+into IP addresses. For most typical use cases, this will be
+sufficient. However, it is possible to pass a custom `lookup`
+function as an option in several places throughout the QUIC API:
+
+* `net.createQuicSocket()`
+* `quicsocket.addEndpoint()`
+* `quicsocket.connect()`
+* `quicsocket.listen()`
+
+The custom `lookup` function must teturn a `Promise` that is
+resolved once the lookup is complete. It will be invoked with
+two arguments:
+
+* `address` {string|undefined} The host name to resolve, or
+  `undefined` if no host name was provided.
+* `family` {number} One of `4` or `6`, identifying either
+  IPv4 or IPv6.
+
+```js
+async function myCustomLookup(address, type) {
+  return resolveTheAddressSomehow(address, type);
+}
```

Comment:
I don’t think this example provides much value – it just repeats the argument types. Even something like a little lookup table that implements this for one or two hosts would already be very helpful, I think :)

Reply:
Yeah, this is largely just a placeholder, I just forgot to add a TODO statement to remind me to get back around to it later. Will add the TODO

Reply:
Added the TODO, marking this resolved for now